activation/atxdb.go

```diff
@@ -157,40 +151,26 @@ func (db *ActivationDb) CalcActiveSetFromView(a *types.ActivationTx) (uint32, er
 //   NodeID, SequenceNumber, PrevATXID, LayerID, StartTick, PositioningATX.
 // - ATX LayerID is NipstLayerTime or more after the PositioningATX LayerID.
 // - The ATX view of the previous epoch contains ActiveSetSize activations
-func (db *ActivationDb) ValidateAtx(atx *types.ActivationTx) error {
+func (db *ActivationDb) SyntacticallyValidateAtx(atx *types.ActivationTx) error {
 	if atx.PrevATXId != *types.EmptyAtxId {
 		prevATX, err := db.GetAtx(atx.PrevATXId)
 		if err != nil {
 			return fmt.Errorf("validation failed: prevATX not found: %v", err)
 		}
+		if prevATX.NodeId.Key != atx.NodeId.Key {
+			return fmt.Errorf("previous ATX belongs to different miner. atx.Id: %v, atx.NodeId: %v, prevAtx.NodeId: %v",
+				atx.ShortId(), atx.NodeId.Key, prevATX.NodeId.Key)
+		}
 		if !prevATX.Valid {
```

Comment:
What is the validity that we check here?
If it's syntactic validation, then I am not sure that we have a reason to store the previous invalid ATX.
If it's contextual validation, then I'm not sure that we should have an error (still open question).

Reply:
Fixed in https://github.com/spacemeshos/go-spacemesh/pull/1085